### PR TITLE
pRPC: Support JSON

### PR DESCRIPTION
**Pull request by kvinwang**
The PR adds support for talking to the pRPC with JSON input/output. This makes it easier for more external tools to connect to pruntime RPC.

Example:
Via GET:
```bash
$ curl localhost:8000/prpc/PhactoryAPI.GetInfo
```
Via POST with a query argument `json`:
```bash
$ curl -d "" localhost:8000/prpc/PhactoryAPI.GetInfo?json
```
Or via POST with the Content-Type of JSON:
```bash
$ curl -d "" -H "Content-Type: application/json" localhost:8000/prpc/PhactoryAPI.GetInfo
```

With any of the above calls, the RPC will respond in JSON format:
```json
{"initialized":false,"registered":false,"genesis_block_hash":null,"public_key":null,"ecdh_public_key":null,"headernum":0,"para_headernum":0,"blocknum":0,"state_root":"","dev_mode":false,"pending_messages":0,"score":0,"gatekeeper":{"role":0,"master_public_key":""},"version":"2.0.1","git_revision":"45d5fb4b66d842d6fdec4fe696f29abe675e389d-dirty","memory_usage":{"rust_used":510350,"rust_peak_used":510350,"total_peak_used":16637444096},"waiting_for_paraheaders":false,"system":null,"can_load_chain_state":true,"safe_mode_level":0,"current_block_time":0}
```

Input arguments can also use JSON. For example:
```bash
curl -d '{"url":"https://httpbin.org/post","method":"POST","body":[],"headers":[]}' localhost:8000/prpc/PhactoryAPI.HttpFetch?json
```
outputs:
```json
{"status_code":200,"headers":[{"name":"date","value":"Thu, 09 Mar 2023 03:45:22 GMT"},{"name":"content-type","value":"application/json"},{"name":"content-length","value":"314"},{"name":"server","value":"gunicorn/19.9.0"},{"name":"access-control-allow-origin","value":"*"},{"name":"access-control-allow-credentials","value":"true"}],"body":[123,10,32,32,34,...]}
```

Also added an mount point `/help` the return the pRPC proto definition:
```
$ curl localhost:8000/help 2>/dev/null | head -n 20

syntax = "proto3";

import "google/protobuf/empty.proto";

package pruntime_rpc;

// The Phactory Runtime service definition.
service PhactoryAPI {
  // Get basic information about Phactory state.
  rpc GetInfo (google.protobuf.Empty) returns (PhactoryInfo) {}

  // Sync the parent chain header
  rpc SyncHeader (HeadersToSync) returns (SyncedTo) {}

  // Sync the parachain header
  rpc SyncParaHeader (ParaHeadersToSync) returns (SyncedTo) {}

```

### CODE REVIEW

File: standalone/pruntime/src/api_server.rs [hunk 1]
```diff
@@ -388,10 +412,9 @@ pub(super) fn rocket_acl(args: &super::Args) -> Option<rocket::Rocket<impl Phase
         .merge(("port", public_port))
         .merge(("limits", Limits::new().limit("json", 100.mebibytes())));
 
-    let mut server_acl =
-        rocket::custom(figment).mount("/", routes![getinfo, get_contract_info, get_cluster_info]);
+    let mut server_acl = rocket::custom(figment).mount("/", routes![getinfo, help]);
```
Comment (h4x3rotab) [inline standalone/pruntime/src/api_server.rs:415]: Why `get_contract_info` and `get_cluster_info` was removed, and `help` was added?

**Review by h4x3rotab — APPROVED** [review]
Do you think it's a good idea to copy the description of this PR to some docs as well?

File: standalone/pruntime/src/api_server.rs (hunk 1 above)
Comment (kvinwang) [reply standalone/pruntime/src/api_server.rs:415]: I previously added `get_contract_info` and `get_cluster_info` as the JSON version of `/prpc/PhactoryAPI.GetContractInfo` and `/prpc/PhactoryAPI.GetClusterInfo`. Nobody actually uses it so far. Now that the pRPC itself supports JSON now, it is meaningless to keep them there.

**Comment by kvinwang** [comment]
> Do you think it's a good idea to copy the description of this PR to some docs as well?

Good. Let me add it to `/docs/rpc.md`.

**Comment by kvinwang** [comment]
Added some doc in the latest commit. @shelvenzhou Shelven, do you want to mention it somewhere on the wiki pages?

**Pull request merged**